scene/gui/popup_menu.cpp

```diff
@@ -332,35 +349,44 @@ int PopupMenu::_get_mouse_over(const Point2 &p_over) const {
 }
 
 void PopupMenu::_activate_submenu(int p_over, bool p_by_keyboard) {
-	Popup *submenu_popup = items[p_over].submenu;
+	PopupMenu *submenu_popup = items[p_over].submenu;
 	if (submenu_popup->is_visible()) {
 		return; // Already visible.
 	}
 
+	float win_scale = get_content_scale_factor();
+
+	Point2 panel_ofs_start = Point2(panel->get_offset(SIDE_LEFT), panel->get_offset(SIDE_TOP)) * win_scale;
+	Point2 panel_ofs_end = Point2(panel->get_offset(SIDE_RIGHT), panel->get_offset(SIDE_BOTTOM)).abs() * win_scale;
+
 	Point2 this_pos = get_position();
+	this_pos.y += panel_ofs_start.y + theme_cache.panel_style->get_margin(SIDE_TOP) * win_scale;
 	Rect2 this_rect(this_pos, get_size());
 
 	float scroll_offset = control->get_position().y;
-	float scaled_ofs_cache = items[p_over]._ofs_cache * get_content_scale_factor();
-	float scaled_height_cache = items[p_over]._height_cache * get_content_scale_factor();
+	float scaled_ofs_cache = items[p_over]._ofs_cache * win_scale;
+	float scaled_height_cache = items[p_over]._height_cache * win_scale;
```

Comment:
```suggestion
	const float scaled_ofs_cache = items[p_over]._ofs_cache * win_scale;
	const float scaled_height_cache = items[p_over]._height_cache * win_scale;
```
While you're at it the line above as well, but GH wouldn't let me apply  the suggestion :)